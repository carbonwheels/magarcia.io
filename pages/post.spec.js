import React from 'react';
import renderer from 'react-test-renderer';
import Post from './post';

it('renders correctly', () => {
  const post = {
    id: 'id',
    slug: 'slug',
    title: 'title',
    content: 'content of the file',
    metadata: {
      url: 'http://example.com',
      intro: 'this is the intro',
      date: '2019-13-01',
      layout: 'post',
      tags: ['tag1', 'tag2'],
      readingTime: '5 mins'
    }
  };
  const tree = renderer.create(<Post {...post} />).toJSON();
  expect(tree).toMatchSnapshot();
});
