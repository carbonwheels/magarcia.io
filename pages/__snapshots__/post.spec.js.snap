// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <article
    className="article"
  >
    <h1
      className="contentTitle"
    >
      title
    </h1>
    <div
      className="postMetaInline "
    >
      <time
        dateTime={null}
      >
        Invalid date
      </time>
      <span
        className="middotDivider"
      />
      <span
        className="readingTime"
        title="5 mins"
      />
    </div>
    <div
      className="postContent"
    >
      <p>
        content of the file
      </p>
    </div>
  </article>,
  <aside
    className="aside  "
  >
    <a
      aria-label="Share on Twitter"
      href="https://twitter.com/intent/tweet?text=title&url=https://magarcia.github.iohttp://example.com&via=martinprins"
      rel="noopener noreferrer"
      target="_blank"
      title="Share on Twitter"
    >
      <img
        alt="Twitter"
        height={24}
        src="test-file-stub"
        width={24}
      />
    </a>
    <a
      aria-label="Share on Facebook"
      href="https://www.facebook.com/sharer/sharer.php?u=https://magarcia.github.iohttp://example.com"
      rel="noopener noreferrer"
      target="_blank"
      title="Share on Facebook"
    >
      <img
        alt="Facebook"
        height={24}
        src="test-file-stub"
        width={24}
      />
    </a>
    <a
      aria-label="Share on LinkedIn"
      href="https://www.linkedin.com/shareArticle?mini=true&url=https://magarcia.github.iohttp://example.com&title=title&source=https%3A//www.linkedin.com/in/martingarciamonterde/"
      rel="noopener noreferrer"
      target="_blank"
      title="Share on LinkedIn"
    >
      <img
        alt="LinkedIn"
        height={24}
        src="test-file-stub"
        width={24}
      />
    </a>
  </aside>,
]
`;
